### PR TITLE
dep; update travis, dist, test, coverage

### PR DESCRIPTION
**Pull request by judwhite**
Similar to https://github.com/nsqio/go-nsq/pull/218

Ready for review. @mreiferson @jehiah @ploxiln

### CODE REVIEW

File: dist.sh [hunk 1]
```diff
@@ -18,10 +18,7 @@ set -e
 DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
 rm -rf   $DIR/dist/docker
 mkdir -p $DIR/dist/docker
-rm -rf   $DIR/.godeps
-mkdir -p $DIR/.godeps
-export GOPATH=$DIR/.godeps:$GOPATH
-GOPATH=$DIR/.godeps gpm install
```
Comment (mreiferson) [inline dist.sh]: I've always liked that `dist.sh` preserves the state of my `GOPATH`, should we maintain that property?

Comment (judwhite) [reply dist.sh]: @mreiferson With `dep` you don't have to change `GOPATH`, it uses `./vendor`. Maybe I misunderstood.

Comment (mreiferson) [reply dist.sh]: Perfect!

**Comment by judwhite** [comment]
@mreiferson Squashed. `go1.9 GOARCH=386` failed on `TestReconfigure`, flaky test?

Tests also take a bit to execute, let me know if you want to drop `go1.7`.

**Comment by ploxiln** [comment]
I re-started that one sub-test and it passed. (I intended to copy the failure details and make an issue for the flaky test, but wasn't being careful and lost the copy.)

**Comment by mreiferson** [comment]
this is cool, thanks.

@judwhite last request, update the docs on http://nsq.io/deployment/installing.html#building-from-source?

**Pull request merged**